src/libostree/ostree-repo-pull.c

```diff
@@ -1498,6 +1501,114 @@ ostree_verify_unwritten_commit (OtPullData                 *pull_data,
     }
 #endif /* OSTREE_DISABLE_GPGME */
 
+  if (pull_data->sign_verify)
+    {
+      /* Nothing to check if detached metadata is absent */
+      if (detached_metadata == NULL)
+        {
+          g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                               "Can't verify commit without detached metadata");
+          return FALSE;
+        }
+      /* Shouldn't happen, but see comment in process_verify_result() */
+      if (g_hash_table_contains (pull_data->verified_commits, checksum))
+        return TRUE;
+
+      gboolean ret = FALSE;
+      g_autoptr(GBytes) signed_data = g_variant_get_data_as_bytes (commit);
+      /* list all signature types in detached metadata and check if signed by any? */
+      g_auto(GStrv) names = ostree_sign_list_names();
+      for (guint i=0; i < g_strv_length (names); i++)
+        {
+          g_autoptr (OstreeSign) sign = NULL;
+          g_autoptr (GVariant) signatures = NULL;
+          g_autofree gchar *signature_key = NULL;
+          g_autofree GVariantType *signature_format = NULL;
+          g_autofree gchar *pk_ascii = NULL;
+          g_autofree gchar *pk_file = NULL;
+
+          if ((sign = ostree_sign_get_by_name (names[i], error)) == NULL)
+          {
+              g_clear_error (error);
```

Comment:
This error will likely never happen, but maybe you could use a local `error` variable instead

Reply:
good, thanks.